ql/src/java/org/apache/hadoop/hive/ql/udf/UDFSubstr.java

```diff
@@ -64,6 +65,27 @@ public UDFSubstr() {
     r = new Text();
   }
 
+  public Text evaluate(Text t, LongWritable pos, LongWritable len) {
+
+    if ((t == null) || (pos == null) || (len == null)) {
+      return null;
+    }
+
+    r.clear();
+    if ((len.get() <= 0)) {
+      return r;
+    }
+
+    String s = t.toString();
+    int[] index = makeIndex(pos.get(), len.get(), s.length());
+    if (index == null) {
+      return r;
+    }
+
+    r.set(s.substring(index[0], index[1]));
+    return r;
+  }
```

Comment:
Java String accepts int parameters only
https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#substring-int-int-

How about creating a common `evaluate` which receives int parameters and both the `LongWritable` and `IntWritable` version call it?

```
public Text evaluate(Text t, IntWritable pos, IntWritable len) {

    if ((t == null) || (pos == null) || (len == null)) {
      return null;
    }

   return evaluate(t, pos.get(), len.get());
}
  
public Text evaluate(Text t, LongWritable pos, LongWritable len) {

    if ((t == null) || (pos == null) || (len == null)) {
      return null;
    }

   // Handle long int overflow
   return evaluate(t, (int)pos.get(), (int)len.get());
}

private Text evaluate(Text t, int pos, int len) {
...
}
```